### PR TITLE
[Android] Fix IllegalStateException crash in WebSocketModule.java

### PR DESCRIPTION
**Pull request by kevinstumpf**
- Motivation: The WebSocket implementation on Android crashes the app when an attempt is made to write on a web socket that was closed due to a spotty connection. We found this issue by using Pusher, which is built on WebSockets. The following stack trace reveals that the WebSocketModule doesn't catch the case of a closed connection, when a consumer attempts to write:
```sh
Fatal Exception: java.lang.IllegalStateException: closed
       at com.squareup.okhttp.internal.ws.RealWebSocket.sendMessage(RealWebSocket.java:109)
       at com.facebook.react.modules.websocket.WebSocketModule.send(WebSocketModule.java:176)
       at java.lang.reflect.Method.invoke(Method.java)
       at java.lang.reflect.Method.invoke(Method.java:372)
       at com.facebook.react.bridge.BaseJavaModule$JavaMethod.invoke(BaseJavaModule.java:249)
       at com.facebook.react.bridge.NativeModuleRegistry$ModuleDefinition.call(NativeModuleRegistry.java:158)
       at com.facebook.react.bridge.NativeModuleRegistry.call(NativeModuleRegistry.java:58)
       at com.facebook.react.bridge.CatalystInstanceImpl$NativeModulesReactCallback.call(CatalystInstanceImpl.java:428)
       at com.facebook.react.bridge.queue.NativeRunnableDeprecated.run(NativeRunnableDeprecated.java)
       at android.os.Handler.handleCallback(Handler.java:739)
       at android.os.Handler.dispatchMessage(Handler.java:95)
       at com.facebook.react.bridge.queue.MessageQueueThreadHandler.dispatchMessage(MessageQueueThreadHandler.java:31)
       at android.os.Looper.loop(Looper.java:145)
       at com.facebook.react.bridge.queue.MessageQueueThreadImpl$3.run(MessageQueueThreadImpl.java:184)
       at java.lang.Thread.run(Thread.java:818)
```

- Test Plan: Create a WebSocket connection and attempt to write after the connection became unavailable due to a spotty connection. An easy way to do this is to establish a Pusher connection with [pusher-websocket-iso/react-native](https://github.com/pusher-community/pusher-websocket-js-iso#react-native) and fake or force a spotty connection.

### CODE REVIEW

File: ReactAndroid/src/main/java/com/facebook/react/modules/websocket/WebSocketModule.java [hunk 1]
```diff
@@ -178,6 +179,8 @@ public void send(String message, int id) {
         new Buffer().writeUtf8(message));
     } catch (IOException e) {
       notifyWebSocketFailed(id, e.getMessage());
+    } catch (IllegalStateException e) {
+      notifyWebSocketFailed(id, e.getMessage());
```
Comment (satya164) [inline ReactAndroid/src/main/java/com/facebook/react/modules/websocket/WebSocketModule.java]: Please combine these two catch blocks like this, `(IOException | IllegalStateException e)`

**Comment by kevinstumpf** [comment]
@satya164 thanks for the quick review. Made the change.

**Comment by satya164** [comment]
Thank you. Will merge as soon as the tests pass :)

**Comment by satya164** [comment]
@facebook-github-bot shipit

**Pull request closed without merging**